add-alfresco-theme.sh

```diff
@@ -0,0 +1,85 @@
+#!/bin/bash -e
+
+log_info() {
+    echo "info::: $1"
+}
+
+log_error() {
+    echo "error::: $1"
+    exit 1
+}
+
+print_usage() {
+    cat <<EOF
+    The required environment variables are not set:
+    - THEME_VERSION
+    OR
+    - GIT_REPO
+    - GIT_BRANCH ('master' will be used if not set)
+
+    Example-1, adding alfresco-theme from a branch:
+        export GIT_REPO=alfresco-keycloak-theme
+        export GIT_BRANCH=test-branch
+        sh add-alfresco-theme.sh
+
+    Example-2, adding alfresco-theme from a released version:
+        export THEME_VERSION=0.1
+        sh add-alfresco-theme.sh
+EOF
+    exit 1
+}
+
+if [ -z "$THEME_VERSION" ] && [ -z "$GIT_REPO" ]; then
+    print_usage
+fi
+
+WORK_DIR="$PWD"
+TMP=$(mktemp -d)
+# Get keycloak version
+log_info "Getting keycloak version..."
+VERSION=$(mvn help:evaluate -Dexpression=project.version -q -DforceStdout)
+log_info "Found keycloak version: $VERSION"
+
+KEYCLOAK_DIR="$WORK_DIR/distribution/server-dist/target"
+KEYCLOAK_DIST="$KEYCLOAK_DIR/keycloak-$VERSION.zip"
+if [ ! -f "$KEYCLOAK_DIST" ]; then
+    log_error "$KEYCLOAK_DIST does not exist."
+fi
+
+cd $TMP
+if [ -n "$GIT_REPO" ]; then
+    if [ -z "$GIT_BRANCH" ]; then
+        GIT_BRANCH='master'
+    fi
+
+    # Clone repository
+    log_info "Clone branch '$GIT_BRANCH' from repo: $GIT_REPO"
+    git clone --depth 1 https://github.com/Alfresco/$GIT_REPO.git -b $GIT_BRANCH alfresco-keycloak-theme
+
+    mkdir alfresco
+    log_info "Copy Alfresco theme..."
+    cp -rf alfresco-keycloak-theme/theme/* alfresco/
+else
+    THEME_DIST="https://github.com/Alfresco/alfresco-keycloak-theme/releases/download/$THEME_VERSION/alfresco-keycloak-theme-$THEME_VERSION.zip"
+    log_info "Download Alfresco theme from: $THEME_DIST"
+
+    curl -sSLO "$THEME_DIST"
+    if ! unzip -oq alfresco-keycloak-theme-$THEME_VERSION.zip; then
+        log_error "Couldn't unzip alfresco-keycloak-theme."
+    fi
+fi
+
+# unzip keycloak in the current directory (i.e. TMP)
+log_info "Unzipping 'keycloak-$VERSION'..."
+unzip -oq $KEYCLOAK_DIST -d .
```

Comment:
Should we consider adding the theme to the zipped distribution rather than unzipping, adding, then re-zipping?

https://unix.stackexchange.com/questions/161905/adding-unzipped-files-to-a-zipped-folder

Reply:
I actually tried that (as you mentioned in the link above), but didn't work as we don't have access to the source files.
In order for the update to work, I have to create the same folder structure and etc. which I think it could be more error prone.

Reply:
Yes, I assumed we still need to unzip the theme, but at least we wouldn't have to unzip the distribution which seems to be > 500 MB uncompressed.

You're saying you'd need to create the folder structure of where the theme needs to go in the distribution zip?

Reply:
Yes, that's what I meant: `unzip keycloak distribution -> copy alfresco theme -> update the zip file`, but even with this approach, I was getting **_zip warnings_**, so that's why I opted for unzip, copy and create zip steps.